File: find_string/find_consecutive.cpp
#include <iostream>
// Find substring with highest consecutive repetition
using namespace std;

string findConsecutiveRepeat(string input_string)
{
    // TODO: validate input
    string top_match = "";
    int max_match = 0, current_match = 0;
    char prev = '\0';
    for (char &c : input_string)
    {
        if (prev == c)
        cout << c;
    }
    return "dddd";
}

int main()
{
    string input = "abcddddefefefefef";
    string actual = findConsecutiveRepeat(input),
           expected = "dddd";

    // Assert
    cout << actual << "\n";
    cout << ((actual == expected) ? "." : "E");
    cout << "\n\n";
}

File: find_string/find_consecutive.py
input_string = "abbdfghhhhhhkjkjkjkjkjkkkkkkkj"

prev = ''
top = ''
count = 0
max = 0
for x in input_string:
    if prev == x:
        count += 1
        if count > max:
            max = count
            top = x
    else:
        count = 1
    prev=x
print(f"From {input_string} we found the longest consecutive repeated string:")
print(f"-{top*max}- ")
